lib/sidekiq/throttled/patches/super_fetch.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require "sidekiq"
+
+module Sidekiq
+  module Throttled
+    module Patches
+      module SuperFetch
+        # Retrieves job from redis.
+        #
+        # @return [Sidekiq::Throttled::UnitOfWork, nil]
+        def retrieve_work
+          work = super
+
+          if work && Throttled.throttled?(work.job)
+            Throttled.cooldown&.notify_throttled(work.queue)
+            requeue_throttled(work)
+            return nil
+          end
+
+          Throttled.cooldown&.notify_admitted(work.queue) if work
+
+          work
+        end
+
+        private
+
+        # Pushes job back to the head of the queue, so that job won't be tried
+        # immediately after it was requeued (in most cases).
+        #
+        # @note This is triggered when job is throttled. So it is same operation
+        #   Sidekiq performs upon `Sidekiq::Worker.perform_async` call.
+        #
+        # @return [void]
+        def requeue_throttled(work)
+          if work.respond_to?(:local_queue)
+            # if a SuperFetch UnitOfWork, SuperFetch will requeue it using lpush
+            work.requeue
+          else
+            # Fallback to BasicFetch behavior
+            redis { |conn| conn.lpush(work.queue, work.job) }
+          end
```

Comment:
Been thinking about requeue logic in context of #150 and I think default logic should simply take unit_of_work's job and push it using `Sidekiq::Client.push(message)`

Reply:
Either way, that can be done separately (when I will have time to work on #150)

Reply:
I think it makes sense to change the behavior separately, keeping it in sync with the behavior for BasicFetch. 

One caveat with using `Sidekiq::Client.push(message)` instead of `conn.lpush` is that it will overwrite the original `enqueued_at` https://github.com/sidekiq/sidekiq/blob/7df28434f03fa1111e9e2834271c020205369f94/lib/sidekiq/client.rb#L260

Reply:
The original `enqueued_at` value isn't really important to us though, so we'd be fine if it got replaced with each throttle

Reply:
> The original `enqueued_at` value isn't really important to us though, so we'd be fine if it got replaced with each throttle

It might be important for sidekiq-pro expiring jobs.

Reply:
Oh. I missed that it replaces `enqueued_at`. Let's keep it as is - as long as we can guarantee that it won't break a reliable (or whatever it is called) client. In either case, we need to call acknowledgement so that the job gets removed from the temporary queue.

Reply:
The `requeue` method of the SuperFetch UnitOfWork takes care of removing from the temporary queue, so I'll add that it does in a comment.

@freemanoid the expiring jobs middleware doesn't use `enqueued_at`